### PR TITLE
[Fluent] Traffic Manager fluent interfaces and implementation

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anuchandy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (anuchan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/TrafficManager/Microsoft.Azure.Management.TrafficManager.Fluent/TrafficManager.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.Azure.Management.Resource.Fluent.Authentication;
+using Microsoft.Azure.Management.Resource.Fluent.Core;
```
Comment (hovsepm) [inline src/ResourceManagement/TrafficManager/Microsoft.Azure.Management.TrafficManager.Fluent/TrafficManager.cs]: Got you here! :))) License header.

Comment (anuchandy) [inline src/ResourceManagement/TrafficManager/Microsoft.Azure.Management.TrafficManager.Fluent/TrafficManager.cs]: again :) .. will fix this.

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/gulpfile.js [hunk 2]
```diff
@@ -71,7 +71,26 @@ var mappings = {
         'source': 'arm-cdn/2016-10-02/swagger/cdn.json',
         'package': 'Microsoft.Azure.Management.Cdn.Fluent',
         'args': '-FT 2'
+    },
+    'cdn': {
+        'dir': 'Cdn/Microsoft.Azure.Management.Cdn.Fluent',
```
Comment (hovsepm) [inline src/ResourceManagement/gulpfile.js]: Please remove the CDN entry that you've added. It already exists on line 70.

File: src/ResourceManagement/Azure.Fluent/Fluent.Tests/TrafficManager/TrafficManagerTests.cs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+                int c = 0;
+                foreach (var endpoint in profile.ExternalEndpoints.Values)
+                {
+                    Assert.Equal(EndpointType.EXTERNAL, endpoint.EndpointType);
+                    if (endpoint.Name.Equals(externalEndpointName21, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Assert.Equal(1, endpoint.RoutingPriority);
+                        Assert.Equal(externalFqdn21, endpoint.Fqdn);
+                        Assert.NotNull(endpoint.MonitorStatus);
+                        // Assert.Equal(Region.US_EAST, endpoint.SourceTrafficLocation);
```
Comment (hovsepm) [inline src/ResourceManagement/Azure.Fluent/Fluent.Tests/TrafficManager/TrafficManagerTests.cs]: > // Assert.Equal(Region.US_EAS [](start = 23, length = 30)

why is this Assert commented?

File: src/ResourceManagement/Azure.Fluent/Fluent.Tests/TrafficManager/TrafficManagerTests.cs [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+                        c++;
+                    }
+                    else if (endpoint.Name.Equals(externalEndpointName22, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Assert.Equal(2, endpoint.RoutingPriority);
+                        Assert.Equal(externalFqdn22, endpoint.Fqdn);
+                        Assert.NotNull(endpoint.MonitorStatus);
+                        // Assert.Equal(Region.US_EAST2, endpoint.SourceTrafficLocation);
```
Comment (hovsepm) [inline src/ResourceManagement/Azure.Fluent/Fluent.Tests/TrafficManager/TrafficManagerTests.cs]: >// [](start = 24, length = 2)

same question "why is this Assert commented?"

File: src/ResourceManagement/Azure.Fluent/Microsoft.Azure.Management.Fluent/project.json [hunk 5]
```diff
@@ -26,7 +26,8 @@
     "Microsoft.Azure.Management.KeyVault.Fluent": "[1.0.0-beta3, 2.0)",
     "Microsoft.Azure.Management.Network.Fluent": "[1.0.0-beta3, 2.0)",
     "Microsoft.Azure.Management.ResourceManager.Fluent": "[1.0.0-beta3, 2.0)",
-    "Microsoft.Azure.Management.Storage.Fluent": "[1.0.0-beta3, 2.0)"
+    "Microsoft.Azure.Management.Storage.Fluent": "[1.0.0-beta3, 2.0)",
+    "Microsoft.Azure.Management.TrafficManager.Fluent": "1.0.0-beta3"
```
Comment (hovsepm) [inline src/ResourceManagement/Azure.Fluent/Microsoft.Azure.Management.Fluent/project.json]: >Fluent": "1.0.0-beta3" [](start = 47, length = 22)

please fix the version reference by following the same pattern as the libraries above.

**Comment by hovsepm** [comment]
>     "Microsoft.Azure.Management.Dns.Fluent": "1.0.0-beta3",

same here. please use version reference range as we have it for other packages. 

---
Refers to: src/ResourceManagement/Azure.Fluent/Microsoft.Azure.Management.Fluent/project.json:25 in b0609d4. [](commit_id = b0609d4f59839b5f5175fda6aeb7e180fa83b5d7, deletion_comment = False)

**Review by hovsepm — CHANGES_REQUESTED** [review]
Please address above comments

**Review by hovsepm — APPROVED** [review]

File: src/ResourceManagement/Azure.Fluent/Fluent.Tests/TrafficManager/TrafficManagerTests.cs (hunk 3 above)
Comment (anuchandy) [inline src/ResourceManagement/Azure.Fluent/Fluent.Tests/TrafficManager/TrafficManagerTests.cs]: @hovsepm Fixed in the PR https://github.com/Azure/azure-sdk-for-net/pull/2595

**Pull request merged**